pkg/controller/config-controller.go

```diff
@@ -1,377 +1,190 @@
+	storagev1 "k8s.io/api/storage/v1"
 
-	// Bind the route SharedIndexInformer to the route queue
-	c.routeInformer.AddEventHandler(cache.ResourceEventHandlerFuncs{
-		AddFunc: c.handleObject,
-		UpdateFunc: func(old, new interface{}) {
-			newDepl := new.(*routev1.Route)
-			oldDepl := old.(*routev1.Route)
-			if newDepl.ResourceVersion == oldDepl.ResourceVersion {
-				// Periodic resync will send update events for all known routes.
-				return
-			}
-			c.handleObject(new)
-		},
-
-		DeleteFunc: c.handleObject,
-	})
+	extensionsv1beta1 "k8s.io/api/extensions/v1beta1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
 
-	// Bind the CDIconfig SharedIndexInformer to the CDIconfig queue
-	c.configInformer.AddEventHandler(cache.ResourceEventHandlerFuncs{
-		AddFunc: c.handleObject,
-		UpdateFunc: func(old, new interface{}) {
-			newDepl := new.(*cdiv1.CDIConfig)
-			oldDepl := old.(*cdiv1.CDIConfig)
-			if newDepl.ResourceVersion == oldDepl.ResourceVersion {
-				// Periodic resync will send update events for all known CDIconfigs.
-				return
-			}
-			c.handleObject(new)
-		},
+	"github.com/go-logr/logr"
+	cdiv1 "kubevirt.io/containerized-data-importer/pkg/apis/core/v1alpha1"
 
-		DeleteFunc: c.handleObject,
-	})
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
 
-	return c
+// CDIConfigReconciler members
+type CDIConfigReconciler struct {
+	Client                 client.Client
+	Scheme                 *runtime.Scheme
+	Log                    logr.Logger
+	UploadProxyServiceName string
+	ConfigName             string
+	CDINamespace           string
 }
 
-func (c *ConfigController) handleObject(obj interface{}) {
-
-	var object metav1.Object
-	var ok bool
-	if object, ok = obj.(metav1.Object); !ok {
-		tombstone, ok := obj.(cache.DeletedFinalStateUnknown)
-		if !ok {
-			runtime.HandleError(errors.Errorf("error decoding object, invalid type"))
-			return
-		}
-		object, ok = tombstone.Obj.(metav1.Object)
-		if !ok {
-			runtime.HandleError(errors.Errorf("error decoding object tombstone, invalid type"))
-			return
-		}
-		klog.V(3).Infof("Recovered deleted object '%s' from tombstone", object.GetName())
-	}
-	klog.V(3).Infof("Processing object: %s", object.GetName())
+// Reconcile the reconcile loop for the CDIConfig object.
+func (r *CDIConfigReconciler) Reconcile(req reconcile.Request) (reconcile.Result, error) {
+	log := r.Log.WithValues("CDIConfig", req.NamespacedName)
+	log.Info("reconciling CDIConfig")
 
-	var config *cdiv1.CDIConfig
-	config, err := c.configLister.Get(c.configName)
-	if err != nil {
-		runtime.HandleError(errors.Errorf("error getting CDI config: %s", err))
-		return
+	var config cdiv1.CDIConfig = *MakeEmptyCDIConfigSpec(r.ConfigName)
```

Comment:
May make more sense for this variable to be a pointer

Reply:
yeah you are right.